src/backends/tensorrt/plan_backend.cc

```diff
@@ -665,15 +799,37 @@ PlanBackend::Context::Run(std::vector<Scheduler::Payload>* payloads)
 
   // For each input, concatenate input values from each payload into
   // the corresponding binding.
-  for (int bindex = 0; bindex < engine_->getNbBindings(); ++bindex) {
-    if (!engine_->bindingIsInput(bindex)) {
+  for (int bindex = 0; bindex < num_expected_bindings_; ++bindex) {
+    if (!engine_->bindingIsInput(binding_offset_ + bindex)) {
       continue;
     }
 
+    // Get the shape of the input. The provider has already checked
+    // that the request shape is valid so don't need to do it here.
+    std::vector<int64_t> shape;
+
+    // The first element of the vector will be the batch size
+    shape.push_back(total_batch_size);
+
     const std::string& name = engine_->getBindingName(bindex);
-    const size_t batch1_byte_size =
-        byte_sizes_[bindex] / std::max(1, max_batch_size_);
 
+    // Can be further optimized
+    size_t batch1_element_cnt = 1;
+    LOG_INFO << input_request_provider->RequestHeader().DebugString();
+    for (const auto& input : input_request_provider->RequestHeader().input()) {
+      const std::string& this_name = input.name();
+      if (this_name.compare(name) == 0) {
+        for (auto dim : input.dims()) {
+          shape.push_back(dim);
+          batch1_element_cnt *= dim;
+        }
+      }
+    }
+
+    const size_t batch1_byte_size =
+        (byte_sizes_[bindex] /
```

Comment:
The batch1_byte_size is just the batch1_element_cnt times the datatype size... so this is a strange way to calculate.